### DISCUSSION TITLE
Primefaces upgrade to 14 : DefaultCommand with same scope and target.

### DISCUSSION THREAD

**Discussion by pradipt1505** [primefaces/community#3233] [PrimeFaces] [answered]
Hi Team,

We are working on migrating PrimeFaces 11(jakarta) to PrimeFaces 14(jakarta). In our project, we are using multiple defaultCommand in the same form. This works completely fine in PrimeFaces 11. But after migration, we are observing following error:
![image](https://github.com/user-attachments/assets/f4d8e11a-8152-4b8a-9bd1-4868c863ee81)

On further analysis we observed that it is happening when scope and target value is same for defaultCommand. The same logic works properly with Primefaces 11.

On checking the changes done from PrimeFaces 11 to PrimeFaces 14, we observed that there was change done 7 months ago in the same line. In the above case, this.jqTarget[0] is undefined due to which we are getting error.

![image](https://github.com/user-attachments/assets/3e235062-2aad-4a14-8c1e-1c0cf1511be6)

Can you please review and provide your feedback?

**Comment by melloware** [comment]
Can you create a PF test reproducer showing the issue?

**Reply by pradipt1505** [reply]
Hi @melloware : Have created a short reproducer as requested. Please check. Let me know if you need any further details.

https://github.com/pradipt1505/primefaces-test

**Reply by melloware** [reply]
Excellent I will take a look tomorrow.

**Reply by pradipt1505** [reply]
Thanks. Will wait for your response. Please note, after creating reproducer have observed issue is not because of same value in target and scope but when the targetted component for defaultCommand has conditional rendering. So on page load, since the value is not present it throws error.

**Reply by melloware** [reply, marked as answer]
Submitted fix for 14.0.10 to make it work like 11 did: https://github.com/primefaces/primefaces/pull/13089
